File: zad2/CS/PRV_zad2_CS/MatrixFillers/RandomMatrixFiller.cs
using PRV2;

namespace MatrixFillers;

public class RandomMatrixFiller : IMatrixFiller<int>
{
    private Random _random;
    private Func<int> _valueGenerator;

    public RandomMatrixFiller(Func<int> valueGenerator)
    {
            _random = new Random();
            _valueGenerator = valueGenerator;
    }
    public void Fill(Matrix matrix)
    {
        for (int i = 0; i < matrix.Size; i++)
        {
            for (int j = 0; j < matrix.Size; j++)
            {
                matrix[i, j] = _valueGenerator();
            }
        }
    }
}

File: zad2/Java/src/MatrixFillers/RandomMatrixFiller.java
package MatrixFillers;

import java.util.Random;

public class RandomMatrixFiller implements MatrixFiller<Integer> {
    @Override
    public void fill(Matrix matrix) {
        Random r = new Random();
        for (int i = 0; i < matrix.size(); i++){
            for (int j = 0; j < matrix.size(); j++){
                matrix.set(i, j, r.nextInt(0, matrix.size() * matrix.size() + 1));
            }
        }
    }
}
